### ISSUE TITLE
False Positives on Methods/Functions That Return *http.Response

### ISSUE THREAD

**Issue by tsvanharen** [timakin/bodyclose#4]
For functions and methods that return a *http.Response and handle the closing of the response body by the function/method caller, this tool flags the response inside the function/method as needing to be closed, when it is being closed by the caller.

**Comment by dcu** [comment]
sample program:

```go
package main

import (
	"net/http"
)

func get() *http.Response {
	resp, _ := http.Get("https://example.com")
	return resp
}

func main() {
	resp := get()
	resp.Body.Close()
}
```

**Comment by timakin** [comment]
Thank you for reporting this one.
I'll fix it with #3 .

**Comment by timakin** [comment]
Fixed and added as a testcase.
https://github.com/timakin/bodyclose/blob/master/passes/bodyclose/testdata/src/a/issue1.go
